### PR TITLE
add target-mongodb

### PR DESCRIPTION
**Pull request by sbalnojan**
Adding target-mongodb so I can get my blogpost out. @tayloramurphy @pnadolny13 

All Singer definitions are stored in `/_data/taps/` or `/_data/targets`. The minimal requirement for adding a tap or target will match the following format:

## Checklist

- [x] Add/update the file in the appropriate folder (`/taps` or `/targets`). The name of the file should match the name of the tap. If there is already one, add a descriptor to the name such as `-search`.
- [x] Add/update the PNG logo image in `/assets/logos/<taps or targets>`. The image name must match the YAML file name.
- [x] Tag `@tayloramurphy` or `@pnadolny13` to flag it for review. Or post to the [#hub](https://meltano.slack.com/archives/C01UGBSJNG5) channel on Meltano slack.

## Reviewer Checklist

- [ ] Validate file against JSON Schema. https://www.jsonschema.net/home is an option.
- [ ] Build website locally and validate everything works.

### CODE REVIEW

File: _data/meltano/loaders/target-mongodb/hotgluexyz.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+- name: connection_string
+  label: Connection String
+  description: Connection string in the format mongodb://[${MONGO_USERNAME:${MONGO_PASSWORD}@]mongodb0.example.com:PORT
+  kind: string
+- name: db_name
+  label: DB Name
+  description: Name of the target db inside your mongoDB
+  kind: string
+settings_group_validation:
+- []
```
Comment (pnadolny13) [inline _data/meltano/loaders/target-mongodb/hotgluexyz.yml:25]:
```suggestion
- - connection_string
  - db_name
```

@sbalnojan It [looks like](https://gitlab.com/hotglue/target-mongodb/-/blob/main/target_mongodb/target.py#L20) these are required configs.

File: _data/meltano/loaders/target-mongodb/hotgluexyz.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+- meltano_sdk
+- database
+namespace: target_mongodb
+pip_url: git+https://gitlab.com/hotglue/target-mongodb.git
+repo: https://gitlab.com/hotglue/target-mongodb
+settings:
+- name: connection_string
+  label: Connection String
+  description: Connection string in the format mongodb://[${MONGO_USERNAME:${MONGO_PASSWORD}@]mongodb0.example.com:PORT
+  kind: string
```
Comment (pnadolny13) [inline _data/meltano/loaders/target-mongodb/hotgluexyz.yml:19]:
```suggestion
  kind: password
```

@sbalnojan what do you think about making this a `password` type? I think many users would put their user/pass in there directly which would expose creds into the meltano.yml. I found a few other cases like airflow/superset where we allow a connection string as a string type but might be worth considering. [target-postgres](https://github.com/meltano/hub/blob/4eafa14a013b762342d2c5766465b22205c4d999/_data/meltano/loaders/target-postgres/meltano.yml#L45) example

File: _data/meltano/loaders/target-mongodb/hotgluexyz.yml (hunk 1 above)
Comment (sbalnojan) [reply _data/meltano/loaders/target-mongodb/hotgluexyz.yml:25]: true, and thanks to @afolson and her "how to add a tap" guide I now know that this field exists ;)

File: _data/meltano/loaders/target-mongodb/hotgluexyz.yml (hunk 2 above)
Comment (sbalnojan) [reply _data/meltano/loaders/target-mongodb/hotgluexyz.yml:19]: puh, I agree with you. We should advocate for data best practices, and there I believe in need to know and data scarcity. So better err on the cautious side here.

File: _data/meltano/loaders/target-mongodb/hotgluexyz.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+name: target-mongodb
+variant: hotgluexyz
+label: MongoDB
+logo_url: /assets/logos/loaders/mongodb.png
+capabilities:
+- about
+description: MongoDB
```
Comment (tayloramurphy) [inline _data/meltano/loaders/target-mongodb/hotgluexyz.yml:7]:
```suggestion
description: General purpose, document-based, distributed database.
```
So it matches the tap.

File: _data/meltano/loaders/target-mongodb/hotgluexyz.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+namespace: target_mongodb
+pip_url: git+https://gitlab.com/hotglue/target-mongodb.git
+repo: https://gitlab.com/hotglue/target-mongodb
+settings:
+- name: connection_string
+  label: Connection String
+  description: Connection string in the format mongodb://[${MONGO_USERNAME:${MONGO_PASSWORD}@]mongodb0.example.com:PORT
+  kind: password
+- name: db_name
+  label: DB Name
```
Comment (tayloramurphy) [inline _data/meltano/loaders/target-mongodb/hotgluexyz.yml:21]:
```suggestion
  label: Database Name
```

**Review by tayloramurphy — APPROVED** [review]
Will be good to circle back and add more details on this but in the interest of getting it published I'm approving :shipit:

**Pull request merged**